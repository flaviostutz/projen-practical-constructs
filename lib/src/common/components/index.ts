export * from './base-tooling';
export * from './common-target-type';
export * from './common-targets';
export * from './release-tasks';
export * from './publish';
