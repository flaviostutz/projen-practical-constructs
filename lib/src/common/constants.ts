export const PROJEN_VERSION = '0.91.13';
export const NODE_VERSION = '20.16.0';
export const MONOTAG_VERSION = '1.23.0';
