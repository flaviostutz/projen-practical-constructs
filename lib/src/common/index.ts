export * from './components';
export * from './files';
export * from './types';
