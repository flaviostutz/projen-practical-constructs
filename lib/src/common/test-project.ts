import { mkdtempSync } from 'node:fs';
import { tmpdir } from 'node:os';
// eslint-disable-next-line unicorn/import-style
import { join } from 'node:path';

import { Project } from 'projen';

export class TestProject extends Project {
  constructor() {
    const outdir = mkdtempSync(join(tmpdir(), 'test-'));
    super({ name: 'test', outdir });
  }
}
