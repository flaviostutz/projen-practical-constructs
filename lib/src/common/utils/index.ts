export * from './ts2jsii';
export * from './monotag';
