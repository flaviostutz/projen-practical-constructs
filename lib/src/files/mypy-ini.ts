import { FileBase, IResolver, Project } from 'projen';

export class MyPyIniFile extends FileBase {
  constructor(project: Project) {
    super(project, 'mypy.ini');
  }

  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  protected synthesizeContent(_resolver: IResolver): string | undefined {
    return `[mypy]
mypy_path = 
    src
    tests
warn_return_any = True
warn_unreachable = True
warn_redundant_casts = True
warn_unused_ignores = True
pretty = True
cache_dir = .cache/.mypy_cache
`;
  }
}
