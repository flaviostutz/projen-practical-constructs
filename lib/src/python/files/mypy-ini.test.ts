/* eslint-disable no-new */
import { Testing } from 'projen';

import { TestProject } from '../../common/test-project';

import { MyPyIniFile } from './mypy-ini';

describe('MyPyIniFile', () => {
  test('synthesizes default mypy.ini', () => {
    const project = new TestProject();
    new MyPyIniFile(project);
    const output = Testing.synth(project);
    expect(output['mypy.ini']).toMatchSnapshot();
  });
});
