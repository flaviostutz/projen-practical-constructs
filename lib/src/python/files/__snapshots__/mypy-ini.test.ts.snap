// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyPyIniFile synthesizes default mypy.ini 1`] = `
"[mypy]
mypy_path = 
    src
    tests
warn_return_any = True
warn_unreachable = True
warn_redundant_casts = True
warn_unused_ignores = True
pretty = True
cache_dir = .cache/.mypy_cache
"
`;
